ha/factory.go

```diff
@@ -0,0 +1,51 @@
+package ha
+
+import (
+	"fmt"
+	"github.com/cashapp/blip"
+	"sync"
+)
+
+type HAManagerFactory interface {
+	Make(monitor blip.ConfigMonitor) (Manager, error)
+}
+
+type defaultFactory struct {
+}
+
+var df = &defaultFactory{}
```

Comment:
In keeping with other pkgs (iirc):

```go
// Internal factory disabled by default (HA not used).
var f = Disabled
```

Then put the factor `Make` method on `disabled` (so it implements both `Manager` and `ManagerFactory` interfaces).

Then in `Register`, use `sync.Once` like you did in an earlier rev to avoid having locks and such.